### DISCUSSION TITLE
Flutter App Displays Blank White UI on iOS 14 & 15, but Works Fine on Android and Other iOS Versions

### DISCUSSION THREAD

**Discussion by JIT-Surya** [community/community#133323] [Actions]
### Body

I have developed a Flutter app that functions correctly on Android and other iOS versions. However, on iOS 14 and 15, the app displays a blank white screen upon launch. This issue does not occur on other iOS versions or Android devices.

I’ve tested the app on various Android devices and iOS versions, where it works as expected. The problem persists when testing on iOS 14 and 15 simulators. I anticipated that the app would display its main content and UI elements properly across all supported iOS versions.

pubspec.yaml file for plugin details:

name: Demo
description: ""
publish_to: 'none'

version: 1.0.1+1

environment:
  sdk: '>=3.2.2 <4.0.0'

dependencies:
  flutter:
    sdk: flutter
  flutter_adaptive_scaffold: ^0.1.10+1
  cupertino_icons: ^1.0.2
  data_connection_checker_tv: ^0.3.5-nullsafety
  get: ^4.6.6
  intl: ^0.19.0
  fluttertoast: ^8.2.4
  shared_preferences: ^2.2.2
  logger: ^2.1.0
  flutter_screenutil: ^5.9.0
  pin_code_text_field: ^1.8.0
  dots_indicator: ^3.0.0
  table_calendar: ^3.1.0
  carousel_slider: ^4.2.1
  image_picker: ^1.0.7
  swipe_cards: ^2.0.0+1
  firebase_auth: ^5.1.2
  google_sign_in: ^6.2.1
  flutter_rating_bar: ^4.0.1
  permission_handler: ^11.3.1
  http: ^1.2.1
  flutter_easyloading: ^3.0.5
  share_plus: ^9.0.0
  url_launcher: ^6.2.6
  time_range_picker: any
  grouped_list: ^6.0.0
  geocoding: ^3.0.0
  emoji_picker_flutter: ^2.2.0
  file_picker: ^8.0.3
  social_media_recorder: ^1.1.13
  cached_network_image: ^3.3.1
  mapbox_search: ^4.2.2
  geolocator: ^12.0.0
  flutter_debouncer: ^2.0.0
  socket_io_client: ^3.0.0-beta.2
  web_socket_channel: ^3.0.0
  injector: ^3.0.0
  instant: ^0.5.1
  path_provider: ^2.1.3
  audioplayers: ^6.0.0
  flutter_sound: ^9.1.9
  connectivity_plus: ^6.0.3
  audio_waveforms: ^1.0.5
  shimmer: ^3.0.0
  firebase_core: ^3.2.0
  flutter_web_auth: ^0.5.0
  crypto: ^3.0.3
  path: ^1.9.0
  month_picker_dialog: ^4.0.0
  firebase_messaging: ^15.0.3
  flutter_local_notifications: ^17.1.2
  firebase_crashlytics: ^4.0.3
  loading_animation_widget: ^1.2.1
  sign_in_with_apple: ^6.1.1
  country_code_picker: ^3.0.0
  location: ^6.0.2

dev_dependencies:
  rename: ^3.0.2
  flutter_test:
    sdk: flutter
  flutter_lints: ^4.0.0

flutter:
  uses-material-design: true
  assets:
    - assets/images/
    - assets/icons/
    - assets/icons/drawer_icons/
    - assets/icons/chat_icon/
    - assets/icons/nav_icons/
  fonts:
    - family: Quicksand
      fonts:
        - asset: fonts/Quicksand-Regular.ttf
        - asset: fonts/Quicksand-Bold.ttf
        - asset: fonts/Quicksand-Light.ttf
        - asset: fonts/Quicksand-Medium.ttf
        - asset: fonts/Quicksand-SemiBold.ttf
    - family: Montserrat
      fonts:
        - asset: fonts/Montserrat-Black.ttf
        - asset: fonts/Montserrat-Bold.ttf
        - asset: fonts/Montserrat-Medium.ttf
        - asset: fonts/Montserrat-ExtraBold.ttf

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by steroSterka** [comment]
try: 
flutter clean               
flutter pub get
cd ios
pod install
cd ..

**Comment by erenmalkoc** [comment]
The minimum IOS version set in your project might not be compatible with some features used in your app. Ensure that the IOS deployment target is set to a version that supports all the features you are using. You can check or update this in your project's ios/Runner.xcodeproj settings.

**Comment by AnasCodex** [comment]
To troubleshoot the blank white screen issue on iOS 14 and 15, consider the following steps:

1. **Check for Debugging Information**: Run your app in debug mode on an iOS 14 or 15 simulator and check the console for any error messages or warnings. This can provide insights into what might be causing the issue.

2. **Review Platform-Specific Code**: Ensure that any platform-specific code (e.g., iOS-specific implementations) is compatible with iOS 14 and 15.

3. **Update Dependencies**: Ensure all dependencies are up-to-date. Some plugins might have compatibility issues with older iOS versions.
    ```bash
    flutter pub outdated
    flutter pub upgrade
    ```

4. **Check for Deprecations**: Some APIs or methods might have been deprecated or changed in iOS 14 and 15. Review the official documentation for any deprecated features you might be using.

5. **Ensure Correct Deployment Target**: Ensure your deployment target in `ios/Podfile` is set correctly. Sometimes setting it to a higher version can help resolve compatibility issues.
    ```ruby
    platform :ios, '14.0'
    ```

6. **Review AppDelegate and Info.plist**:
   - Ensure `AppDelegate.swift` or `AppDelegate.m` (depending on your project) is correctly set up.
   - Check `Info.plist` for any missing permissions or configurations.

7. **Check for Native Code Issues**: If your app uses native code or plugins that require native integration, ensure these are properly set up and compatible with iOS 14 and 15.

8. **Rebuild the App**: Clean the build and rebuild the app to ensure there are no stale files causing the issue.
    ```bash
    flutter clean
    flutter build ios
    ```

9. **Use WidgetsBinding**: Sometimes, using `WidgetsBinding.instance.addPostFrameCallback` to delay the execution of certain code until after the first frame can help.
    ```dart
    void main() {
      WidgetsFlutterBinding.ensureInitialized();
      runApp(MyApp());
    }
    ```

10. **Profile and Release Modes**: Test your app in both profile and release modes to see if the issue persists.
    ```bash
    flutter build ios --release
    flutter build ios --profile
    ```

11. **Check for UI Issues**: Ensure that there are no widgets or UI elements causing the app to render a blank screen on these specific iOS versions.

If the issue persists after trying these steps, please provide any error logs or additional details that might help in diagnosing the problem further.
